iravelmakina/linear-transformations-ivelmakina
# Refactoring

### Description:
- Refactored the entire project structure to ensure compatibility with the parser and to function seamlessly as one cohesive program.
- Introduced new files `main.py` , `transformations_validation_visualization.py` and `parser_validation.py` to enhance the project's structure.
- Renamed existing files for improved clarity and enhanced user experience.

### Changes:
- Refactored all project files to align with the parser and ensure smooth integration.
- Added `main.py` for better structuring.
- Added `transformations_validation_visualization.py` to facilitate better visualization and validation of transformations.
- Introduced `parser_validation.py` to enhance parsing functionalities and improve overall project structure.
- Renamed files across the project for improved clarity and user-friendliness.

### How to Test:
1. Clone the updated repository to your local machine.
2. Ensure all dependencies are installed.
3. Execute the main program file to verify that the program runs smoothly without any errors.
4. Test each functionality individually to ensure that transformations and parsing work as expected.
5. Use the new visualization and validation functionalities to confirm that they provide the desired output.
6. Ensure that the renaming of files has not affected the functionality and that the program remains intuitive to use.

### Screenshots:
- 
![image](https://github.com/iravelmakina/linear-transformations-ivelmakina/assets/150923968/3021ef95-d708-4199-a6b4-79ac117b2c2b)
![image](https://github.com/iravelmakina/linear-transformations-ivelmakina/assets/150923968/72175473-d4ea-41c2-b37c-c6f1097feef6)
![image](https://github.com/iravelmakina/linear-transformations-ivelmakina/assets/150923968/4acf58cf-7dbe-4122-9375-28ff70daac50)
